src/main/java/core/basesyntax/PurchaseOperation.java

```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        storage.put(transaction.getFruit(), storage.getOrDefault(transaction.getFruit(), 0) - transaction.getQuantity());
```

Comment:
The `handle` method does not check if the result balance is positive before saving it in the storage. Per the checklist, you should throw a `RuntimeException` if the balance is negative to handle the Purchase operation correctly. Consider checking the balance first and then proceeding with the update if the balance is sufficient.